app/services/statement_creator.rb

```diff
@@ -0,0 +1,71 @@
+class StatementCreator
+
+  attr_accessor :order_detail_ids, :errors, :to_statement, :account_statements, :session_user, :current_facility
+
+  def initialize(params)
+    @order_detail_ids = params[:order_detail_ids]
+    @session_user = params[:session_user]
+    @current_facility = params[:current_facility]
+    @errors = []
+    @to_statement = {}
+  end
+
+  def create
+    OrderDetail.transaction do
+      set_order_details_to_statement
+      setup_statement_from_details
+      raise ActiveRecord::Rollback if errors.any?
```

Comment:
It did not return true by default, I had to explicitly state a return value (hence the last line)